src/Orleans.Serialization/Codecs/IntegerCodec.cs

```diff
@@ -561,7 +565,7 @@ public sealed class Int64Codec : IFieldCodec<long>
         public static void WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, long value) where TBufferWriter : IBufferWriter<byte>
         {
             ReferenceCodec.MarkValueField(writer.Session);
-            if (value <= int.MaxValue && value >= int.MinValue)
+            if (value is <= int.MaxValue and >= int.MinValue)
```

Comment:
Could this branch use `WriteVarInt32((int)value)` instead of `WriteVarInt64`? Or is the zig-zag encoding different for each integer size?
Alternatively, could get rid of one of the fallback branches by restructuring the conditions so there's a single call to `WriteVarInt64`.

Reply:
`UInt64Codec` below could definitely use `WriteVarUInt32` instead of `WriteVarUInt64`.